Switch EndpointSlices to discovery.k8s.io/v1

vthapar opened the pull request:
Fixes #1922
Depends on https://github.com/submariner-io/lighthouse/pull/720

Signed-off-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our developer guide: https://submariner.io/development/
2. Ensure you have added the appropriate tests for your PR: https://submariner.io/development/code-review/#test-new-functionality
3. Read the code review guide to ease the review process: https://submariner.io/development/code-review/
4. If the PR is unfinished, mark it as a draft: https://submariner.io/development/code-review/#mark-work-in-progress-prs-as-drafts
5. If you are using CI to debug, use your private fork: https://submariner.io/development/code-review/#use-private-forks-for-debugging-prs-by-running-ci
6. Add labels to the PR as appropriate.

This template is based on the K8s/K8s template:

https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
-->

vthapar committed:
Switch EndpointSlices to discovery.k8s.io/v1

Fixes #1922
Depends On github.com/submariner-io/lighthouse/pull/720

Signed-off-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

skitt commented:
The uninstall UTs are failing; @tpantelis could you take a look?

@vthapar you also need to update the YAMLs.

tpantelis committed:
Fix UTs due K8s 0.21.11 version bump

In the newer controller-runtime version, the fake client actually
implements the finalizer functionality and deletes the object when
the finalizers are removed. So when we check to see if the finalizer
was removed, the object doesn’t exist. Adjust the tests to verify the
object was deleted.

Signed-off-by: Tom Pantelis <tompantelis@gmail.com>

skitt commented:
Thanks @tpantelis!

vthapar committed:
Fix embeddedyamls for 0.21.11 version bump

Signed-off-by: Vishal Thapar <5137689+vthapar@users.noreply.github.com>

tpantelis committed:
Merge branch 'devel' into endpointslices-v1

tpantelis committed:
Merge branch 'devel' into endpointslices-v1
